### PR TITLE
libjpeg: bump msys2 + simplify autotools build & package + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
I believe also that nmake build doesn't honor build type or env vars like CFLAGS from profile.

**Review by intelligide — APPROVED** [review]

File: recipes/libjpeg/all/conanfile.py [hunk 1]
```diff
@@ -64,35 +77,20 @@ def _build_nmake(self):
...
-            config_args.extend(["--enable-shared=yes", "--enable-static=no"])
-        else:
-            config_args.extend(["--enable-shared=no", "--enable-static=yes"])
-
-        if self.settings.os == "Windows":
-            mingw_arch = {
-                "x86_64": "x86_64",
-                "x86": "i686",
-            }
-            build_triplet = host_triplet = "{}-w64-mingw32".format(mingw_arch[str(self.settings.arch)])
```
Comment (madebr) [inline recipes/libjpeg/all/conanfile.py:85]: I guess conan/autotools is able to figure the gnu triplets on its own?

**Review by madebr — APPROVED** [review]

**Comment by madebr** [comment]
> And I still don't think to check self.settings.compiler != "Visual Studio" is the best way to decide whether to install msys or not. :-)

@dmn-star
MSVC doesn't need it.
And by defining `CONAN_BASH_PATH` (or not), you can disable the msys2 build requirement.
Isn't this sufficient for building inside msys2?

File: recipes/libjpeg/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/libjpeg/all/conanfile.py:85]: yes

**Comment by dmn-star** [comment]
@madebr 
When I was trying to cross build for Android on Windows (os=Windows, settings.compiler=clang), setting CONAN_BASH_PATH was my workaround to skip msys2 download. However, I recently stumbled across it again when I was experimenting with clang on Windows (again os=Windows, settings.compiler=clang). Personally, this does not bother me so much. We have completely stopped building for Android on Windows, it's just way too buggy and time consuming. Then we stopped support for mingw and switched completely to clang. And last but not least we switched from conan to vcpkg on Windows. :-(

Maybe I'm wrong, but I would rewrite the lines 
        if self._settings_build.os == "Windows" and self.settings.compiler != "Visual Studio" and \
            not tools.get_env("CONAN_BASH_PATH"):
             self.build_requires("msys2/cci.latest")
 like this         
        if _mingw() and \
            not tools.get_env("CONAN_BASH_PATH"):
             self.build_requires("msys2/cci.latest")
             
          where _mingw() something like this   if self._settings_build.os == "Windows" and self.settings.compiler = "gcc"

**Review by SSE4 — APPROVED** [review]

**Comment by SSE4** [comment]
> I believe also that nmake build doesn't honor env vars like CFLAGS from profile, or what a consumer would expect from build_type.

it should if we use `VisualStudioBuildEnvironment`?

**Comment by SSE4** [comment]
> And I still don't think to check self.settings.compiler != "Visual Studio" is the best way to decide whether to install msys or not. :-)

Visual Studio builds rely on nmake, while MinGW needs MSYS environment with bash, etc.
other compilers (Intel, Clang, etc.) may use either of this (e.g. there is clang-cl which relies on MS stuff, and there is also clang from MinGW as well. it's hard to write a correct condition to satisfy both).

**Comment by SpaceIm** [comment]
> > I believe also that nmake build doesn't honor env vars like CFLAGS from profile, or what a consumer would expect from build_type.
> 
> it should if we use `VisualStudioBuildEnvironment`?

I guess, but `VisualStudioBuildEnvironment` is not used in `libjpeg` recipe.

**Pull request merged**